app/reducers/transaction.js

```diff
@@ -110,26 +110,29 @@ export const transactionError = (event, { error }) => dispatch => {
 }
 
 // Listener for when a new transaction is pushed from the subscriber
-export const newTransaction = (event, { transaction }) => dispatch => {
-  // Fetch new balance
-  dispatch(fetchBalance())
+export const newTransaction = (event, { transaction }) => (dispatch, getState) => {
+  // add the transaction only if we are not already aware of it
+  if (!getState().transactions.find(tx => tx.tx_hash === transaction.tx_hash)) {
+    // Fetch new balance
+    dispatch(fetchBalance())
 
-  dispatch({ type: ADD_TRANSACTION, transaction })
+    dispatch({ type: ADD_TRANSACTION, transaction })
 
-  transaction.received = transaction.amount > 0
+    transaction.received = transaction.amount > 0
 
-  // HTML 5 desktop notification for the new transaction
-  const notifTitle = transaction.received
-    ? 'On-chain Transaction Received!'
-    : 'On-chain Transaction Sent!'
-  const notifBody = transaction.received
-    ? "Lucky you, you just received a new on-chain transaction. I'm jealous."
-    : "Hate to see 'em go but love to watch 'em leave. Your on-chain transaction successfully sent."
+    // HTML 5 desktop notification for the new transaction
+    const notifTitle = transaction.received
+      ? 'On-chain Transaction Received!'
+      : 'On-chain Transaction Sent!'
+    const notifBody = transaction.received
+      ? "Lucky you, you just received a new on-chain transaction. I'm jealous."
+      : "Hate to see 'em go but love to watch 'em leave. Your on-chain transaction successfully sent."
 
-  showNotification(notifTitle, notifBody)
+    showNotification(notifTitle, notifBody)
 
-  // Generate a new address
-  dispatch(newAddress('np2wkh'))
+    // Generate a new address
+    dispatch(newAddress('np2wkh'))
```

Comment:
Not sure if it should be part of this PR but we shouldn't be generating a new address every time right? We should only generate if the tx was a receiving tx. If it was the user sending BTC then they don't need a new address

Reply:
Great idea - fixed in bcc86db